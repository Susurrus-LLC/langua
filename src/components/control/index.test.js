import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import Control from '.'

describe('Control', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Control />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('renders accurately with and without props and children', () => {
    let tree = renderer.create(<Control />).toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderer.create(<Control addedClasses='test' />).toJSON()
    expect(tree).toMatchSnapshot()

    tree = renderer.create(<Control>Test</Control>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
