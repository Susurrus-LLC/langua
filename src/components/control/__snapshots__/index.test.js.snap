// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control renders accurately with and without props and children 1`] = `
<div
  className="control"
/>
`;

exports[`Control renders accurately with and without props and children 2`] = `
<div
  className="control test"
/>
`;

exports[`Control renders accurately with and without props and children 3`] = `
<div
  className="control"
>
  Test
</div>
`;
